sem/utils.py

```diff
@@ -330,3 +331,104 @@ def compute_sensitivity_analysis(
+            components supported by ns-3.
+    """
+    if not log_component:
+        return None
+
+    ret_dict = {}
+    log_level_list = ['error',
+                      'warn',
+                      'debug',
+                      'info',
+                      'function',
+                      'logic']
+    converter = {
+        'error': ['error'],
+        'warn': ['warn'],
+        'debug': ['debug'],
+        'info': ['info'],
+        'function': ['function'],
+        'logic': ['logic'],
+        'all': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        'level_error': ['error'],
+        'level_warn': ['error', 'warn'],
+        'level_debug': ['error', 'warn', 'debug'],
+        'level_info': ['error', 'warn', 'debug', 'info'],
+        'level_function': ['error', 'warn', 'debug', 'info', 'function'],
+        'level_logic': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        'level_all': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        '**': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        '*': None,
+        'prefix_func': None,
+        'prefix_time': None,
+        'prefix_node': None,
+        'prefix_level': None,
+        'prefix_all': None
+    }
+    for component, levels in log_component.items():
+        log_level_complete = set()
+        if (ns3_log_components is not None) and not (component not in ns3_log_components or component != '*'):
+            raise ValueError(
+                "Log component '%s' is not a valid ns-3 log component. Valid log components: \n%ls" % (
+                    component,
+                    ns3_log_components
+                ))
+        for level in levels.split('|'):
+            if level not in converter:
+                raise ValueError("Log level '%s' for component '%s' is not valid"
+                                 % (level,
+                                    component))
+
+            # '*' represents level_all only if it occurs before the first '|'
+            if levels.split('|')[0] == '*':
+                log_level_complete.update(converter['all'])
+                continue
+
+            # Do not update the dictionary if prefixes are mentioned
+            if converter[level] is not None:
+                log_level_complete.update(converter[level])
+
+        # Update log_level_complete if entry for components exists
+        if component in ret_dict:
+            log_level_complete.update([log_level
+                                      for log_level in ret_dict[component].split('|')
+                                      if log_level not in log_level_complete])
+
+        # Sort the log classes for consistency
+        log_level_sorted = [level for level in log_level_list
+                            if level in log_level_complete]
+
+        if component == '*':
+            if ns3_log_components is None:
+                raise ValueError('No list of ns-3 supported log components passed.\n')
```

Comment:
maybe this can already be part of the long condition above, o part of that condition can be moved here

Reply:
The parse_log_component() can be used in two ways. 
- If ns3_log_components is None: It will just check the log_levels and make them consistent with the database format.
- If ns3_log_components is not None: In addition to the above things, it will also check the validity of components passed. So basically, ns3_log_components is optionally required to check the component validity or mandatory if there is a '*' or '**' in the log levels passed. 

Considering this I do not think we can merge this check with the above code.

Reply:
makes sense